spec/two_pair_spec.rb

```diff
@@ -0,0 +1,33 @@
+RSpec.describe TwoPair do
+  describe '.check' do
+    subject { TwoPair.new.check(hand) }
+
+    context 'when hand is a TwoPair' do
+      let(:hand) do
+        [
+          Card.new('4', 'C'),
+          Card.new('4', 'S'),
+          Card.new('3', 'C'),
+          Card.new('3', 'D'),
+          Card.new('12', 'C')
+        ]
+      end
+
+      it { is_expected.to be true }
+    end
+
+    context 'when hand is not a TwoPair' do
+      let(:hand) do
+        [
+          Card.new('4', 'S'), 
+          Card.new('11', 'S'),
+          Card.new('8', 'S'),
+          Card.new('2', 'S'),
+          Card.new('9', 'S')
+        ]
+      end
+
+      it { is_expected.to be false }
+    end
+  end
+end
```

Comment:
Could be worth adding in a spec that takes in a hand with a single pair, not required though.